kedro-datasets/pyproject.toml

```diff
@@ -22,7 +22,7 @@ hdfs-base = ["hdfs>=2.5.8, <3.0"]
 pandas-base = ["pandas>=1.3, <3.0"]
 plotly-base = ["plotly>=4.8.0, <6.0"]
 s3fs-base = ["s3fs>=0.3.0, <0.5"]
-spark-base = ["pyspark>=2.2, <4.0"]
+spark-base = ["pyspark>=2.2, <3.4"]
```

Comment:
Should this be kept for 4.0? This is weird because pyspark3.4.1 is installed in the 3.11 tests now. Maybe because the alias was broken so it's not doing anything